frontend/src/components/VideoCardPage/VideoCardPage.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import Grid from '@material-ui/core/Grid';
+import TopBar from '../TopBar/TopBar';
+import UploadModal from '../UploadModal/UploadModal';
+import './styles.css';
+import VideoCardsByDate from '../VideoCardsByDate/VideoCardsByDate';
+import NavDrawer from '../NavDrawer/NavDrawer';
+import CustomSnackbar from '../CustomSnackbar/CustomSnackbar';
+import DeleteConfirmationDialog from '../DeleteConfirmationDialog/DeleteConfirmationDialog';
+
+const VideoCardPage = ({
+  renderSnackbars,
+  title,
+  size,
+  showNavDrawer,
+  videosInformation,
+  showUploadModal,
+  showDeletionDialog,
+  selectedCardIds,
+  showDeletionError,
+  iconsToShow,
+  iconsToShowOnSelected,
+  handleIconsClick,
+  handleIconsClickOnSelected,
+  sortedVideoDates,
+  handleSelect,
+  handleDateSelect,
+  showDownloadInProgress,
+  handleActionIconClick,
+  toggleNavDrawer,
+  toggleUploadModal,
+  handleVideoDeletion,
+  toggleDeletionDialog,
+  hideDeletionError,
+  deleteForever,
+  children,
+}) => (
+  <>
+    {renderSnackbars()}
+    <Grid
+      className="root"
+      style={{ height: videosInformation.length === 0 ? '100vh' : 'auto' }}
+      container
+      direction="column"
+    >
+      <NavDrawer
+        open={showNavDrawer}
+        onOpen={toggleNavDrawer}
+        onClose={toggleNavDrawer}
+        spaceTaken={size}
+      />
+      <UploadModal show={showUploadModal} onClose={toggleUploadModal} />
+      <DeleteConfirmationDialog
+        open={showDeletionDialog}
+        onConfirm={handleVideoDeletion}
+        onCancel={toggleDeletionDialog}
+        multipleVideos={selectedCardIds.length > 1}
+        deleteForever={deleteForever}
+      />
+      {showDeletionError && (
+        <CustomSnackbar
+          message={
+            selectedCardIds.length === 1
+              ? 'Oops... Something wrong happened. We could not delete your video'
+              : 'Oops... Something wrong happened. Some videos might not have been deleted'
```

Comment:
Wait, so we reuse same messages for deletion and permanent deletion?

Reply:
Changed